packages/commerce-sdk-react/src/hooks/Test/config.ts

```diff
@@ -0,0 +1,121 @@
+type BasketOptions = MergedOptions<Client, ApiOptions<Basket>>
+
+const updateBasketQuery = (
+    customerId: string | null,
+    basketId: string | undefined,
+    newBasket: Basket
+): Pick<CacheUpdate, 'update'> => {
+    if (!basketId) return {}
+    const update: Array<CacheUpdateUpdate<Basket> | CacheUpdateUpdate<CustomerBasketsResult>> = [
+        {
+            queryKey: ['/baskets', basketId, {basketId}],
+            updater: newBasket
+        }
+    ]
+    if (customerId) {
+        const updateCustomerBaskets: CacheUpdateUpdate<CustomerBasketsResult> = {
+            // Since we use baskets from customer basket query, we need to update it for any basket mutation
+            queryKey: ['/customers', customerId, '/baskets', {customerId}],
+            updater: (oldData) => {
+                // do not update if response basket is not part of existing customer baskets
+                if (!oldData?.baskets?.some((basket) => basket.basketId === basketId)) {
+                    return undefined
+                }
+                const updatedBaskets = oldData.baskets.map((basket) => {
+                    return basket.basketId === basketId ? newBasket : basket
+                })
+                return {
+                    ...oldData,
+                    // TODO: Remove type assertion when RAML specs match
+                    baskets: updatedBaskets as CustomerBasketsResult['baskets']
+                }
+            }
+        }
+        update.push(updateCustomerBaskets)
+    }
+    // TODO: This type assertion is so that we "forget" what type the updater uses.
+    // Is there a way to avoid the assertion?
+    return {update} as CacheUpdate
+}
+
+const removeBasketQuery = (
+    options: BasketOptions,
+    basketId?: string
+): Pick<CacheUpdate, 'remove'> => {
+    if (!basketId) return {}
+    return {
+        remove: [and(matchesApiConfig(options), pathStartsWith(['/baskets', basketId]))]
+    }
+}
+
+const invalidateCustomerBasketsQuery = (
+    customerId: string | null,
+    options: BasketOptions
+): Pick<CacheUpdate, 'invalidate'> => {
+    if (!customerId) return {}
+    return {
+        invalidate: [
+            and(matchesApiConfig(options), pathStartsWith(['/customers', customerId, '/baskets']))
+        ]
+    }
+}
+
+const updateBasketFromRequest = (
+    customerId: string | null,
+    options: BasketOptions,
+    response: Basket
+): CacheUpdate => ({
+    ...updateBasketQuery(customerId, options.parameters?.basketId, response),
+    ...invalidateCustomerBasketsQuery(customerId, options)
+})
+
+const updateBasketFromResponse = (
+    customerId: string | null,
+    options: BasketOptions,
+    response: Basket
+): CacheUpdate => ({
+    ...updateBasketQuery(customerId, response.basketId, response),
+    ...invalidateCustomerBasketsQuery(customerId, options)
+})
+
+export const cacheUpdateMatrix: CacheUpdateMatrix<Client> = {
+    addCouponToBasket: updateBasketFromRequest,
```

Comment:
The name is very misleading here, the cache should only be updated from response data in any of the mutation cases on Success. I understand the reason for these functions is the source of getting basketId, but the fn name indicates that it updates the basket from a data in the request, which is not entirely true. 
Can we just have one single fn? Most cases we have basketId in the options, if it is not available, it is likely it is in the response

```
updateBasketCache = (
    customerId: string | null,
    options: BasketOptions,
    response: Basket
): CacheUpdate => ({
    ...updateBasketQuery(customerId, options.parameters?.basketId || response.basketId, response),
    ...invalidateCustomerBasketsQuery(customerId, options)
})
```

Reply:
This PR is a bit outdated by now, in #959 I've changed the names to `updateBasket` and `updateBasketWithResponseBasketId`.